src/rational/mat_q/arithmetic/mul.rs

```diff
@@ -50,6 +52,54 @@ impl Mul for &MatQ {
     }
 }
 
+impl Mul<&MatZ> for &MatQ {
+    type Output = MatQ;
+
+    /// Implements the [`Mul`] trait for [`MatQ`] and [`MatZ`].
+    ///
+    /// [`Mul`] is implemented for any combination of owned and borrowed [`MatQ`] and [`MatZ`].
+    ///
+    /// Parameters:
+    /// - `other`: specifies the value to multiply with `self`
+    ///
+    /// Returns the product of `self` and `other` as a [`MatQ`].
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::MatQ;
+    /// use qfall_math::integer::MatZ;
+    /// use std::str::FromStr;
+    ///
+    /// let a = MatQ::from_str("[[2/3,1/2],[8/4,7]] mod 3").unwrap();
+    /// let b = MatZ::from_str("[[1,0],[0,1]]").unwrap();
+    ///
+    /// let c = &a * &b;
+    /// let d = a * b;
+    /// let e = &c * d;
+    /// let f = c * &e;
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Panics if the dimensions of `self` and `other` do not match for multiplication.
+    fn mul(self, other: &MatZ) -> Self::Output {
+        if self.get_num_columns() != other.get_num_rows() {
+            panic!("Tried to multiply matrices with mismatching matrix dimensions.");
+        }
+
+        let other_matq = MatQ::from(other);
+        let mut new = MatQ::new(self.get_num_rows(), other.get_num_columns()).unwrap();
+        unsafe { fmpq_mat_mul(&mut new.matrix, &self.matrix, &other_matq.matrix) };
+        new
```

Comment:
Please use `fmpq_mat_mul_fmpz_mat`, as we do not have to clone the entire matrix this way.
```suggestion
        let mut new = MatQ::new(self.get_num_rows(), other.get_num_columns()).unwrap();
        unsafe { fmpq_mat_mul_fmpz_mat(&mut new.matrix, &self.matrix, &other.matrix) };
        new
```

Reply:
I did not know this exists...